File: ck-editor5-image-upload-master/Client/src/service/ProductService.js
import axios from 'axios';

const PRODUCT_REST_API ="http://localhost:8080/product";

class ProductService{

    getProduct(page,size){
        return axios.get("http://localhost:8080/product/?page="+page+"&size="+size);
    }
    getProductAll(){
        return axios.get("http://localhost:8080/productall");
    }
    getProductbyCate(id,page,size){
        return axios.get("http://localhost:8080/productbycategory/"+id+"?page="+page+"&size="+size);
    }
    getProductSale(page,size){
        return axios.get("http://localhost:8080/productsale/?page="+page+"&size="+size);
    }
    getProductSoon(page,size){
        return axios.get("http://localhost:8080/productsoon/?page="+page+"&size="+size);
    }
    getProductTop(page, size ){
        return axios.get("http://localhost:8080/producttop/?page="+page+"&size="+size); 
    }
    getProductBot(page, size){
        return axios.get("http://localhost:8080/productbot/?page="+page+"&size="+size);
    }
    getProductRelated(id){
        return axios.get("http://localhost:8080/productcate/"+id);
    }
    createProduct(id,product){
        return axios.post(PRODUCT_REST_API+'/'+id,product);
    }
    getProductById(id){
        return axios.get(PRODUCT_REST_API+'/'+id);
    }
    deleteProduct(id){
        return axios.delete(PRODUCT_REST_API+'/'+id);
    }
    updateProduct(product,id,idcate){
        return axios.put(PRODUCT_REST_API+'/'+id+"/"+idcate,product);
    }
    updatenews(news,id){
        return axios.put("http://localhost:8080/productnews/"+id,news);
    }
    updateprolated(productrelated,id){
        return axios.put("http://localhost:8080/productrelated/"+id,productrelated);
    }
    updateimages(productimages,id){
        return axios.put("http://localhost:8080/productimage/"+id,productimages);
    }
    findByName(name) {
        return axios.get(PRODUCT_REST_API+'/'+`search?name=${name}`);
    }
    setNews(id,news)
    {
             return axios.post("http://localhost:8080/listnews/"+id,news);
    }
    setProductRelated(id,productrelated)
    {
             return axios.post("http://localhost:8080/listprorelated/"+id,productrelated);
    }
    getAccessSale(page,size){
        return axios.get("http://localhost:8080/accesssale/?page="+page+"&size="+size);
    }
    getAccess(page,size){
        return axios.get("http://localhost:8080/access/?page="+page+"&size="+size);
    }
    getAllAccess(page,size){
        return axios.get("http://localhost:8080/accessall/?page="+page+"&size="+size);
    }
}
export default new ProductService();

File: daoduytinhjavaweb/src/main/java/com/daoduytinh/service/ProductService.java
package com.daoduytinh.service;

import java.util.List;
import java.util.Map;

import javax.validation.Valid;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.Query;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.multipart.MultipartFile;

import com.daoduytinh.entity.News;
import com.daoduytinh.entity.Product;
import com.daoduytinh.entity.ProductImages;
import com.daoduytinh.entity.ProductRelated;
import com.daoduytinh.entity.Salient;
import com.daoduytinh.exception.ResourceNotFoundException;


public interface ProductService {
	public Page<Product> getAllAccess(Pageable pageable);

	public Page<Product> getProductSale(Pageable pageable);
	
	public Page<Product> getAccessSale(Pageable pageable);
	
	public Page<Product> getProductbycategory(@PathVariable(value = "id") Long id,Pageable pageable);
	
	public Page<Product> getProductTop(Pageable pageable);
	
	public Page<Product> getProductSoon(Pageable pageable);
	
	public Page<Product> getProductBot(Pageable pageable);
	
	public Page<Product> getAllProductSort(Pageable pageable);
	
	public Iterable<Product> getAllProduct();
	
	public List<Product> getAllProductCategory(@PathVariable(value = "id") Long id);
	
	public ResponseEntity<Product> getProductById(@PathVariable(value = "id") Long Id) throws ResourceNotFoundException;
	
	public Page<Product> getAccess(Pageable pageable);
	
	 public Product createProduct(@PathVariable(value = "id") Long Id,@RequestBody Product product);
	 
	 public ResponseEntity<?> setProductImage(@PathVariable(value = "id") Long Id,@RequestBody List<ProductImages> images);
	 
	 public ResponseEntity<?> setProductNews(@PathVariable(value = "id") Long Id,@RequestBody List<News> news);
	 
	 public ResponseEntity<?> setProductrelated(@PathVariable(value = "id") Long Id,@RequestBody List<ProductRelated> productRelateds);
	 
	 public ResponseEntity<Product> updateProduct(@PathVariable(value = "id") Long Id,@PathVariable(value = "idcate") Long Idcate,
	         @Valid @RequestBody Product productDetails) throws ResourceNotFoundException;
	 
	 public Map<String, Boolean> deleteProduct(@PathVariable(value = "id") Long Id)
	         throws ResourceNotFoundException;
	 
	 public ResponseEntity<List<Product>> getAllSearch(@RequestParam(required = false) String name);
	 
	 public ResponseEntity<?> uploadImage(@RequestParam("imageFile") MultipartFile file);


}
